### PR TITLE
EVG-7732 add virtual workstation commands to project page

### PR DESCRIPTION
**Pull request by ablack12**
Breaking this ticket into steps so this would be step one: add workstation commands to the project page.
(Not sure how to make directory match textarea width / if it should)
image

### CODE REVIEW

File: public/static/js/projects.js [hunk 1]
```diff
@@ -477,6 +478,9 @@ mciModule.controller('ProjectCtrl', function ($scope, $window, $http, $location,
     if ($scope.patch_alias) {
       $scope.addPatchAlias();
     }
+    $scope.settingsFormData.workstation_config = _.filter($scope.settingsFormData.workstation_config, function (d) {
```
Comment (john-m-liu) [inline public/static/js/projects.js:481]: If you're doing validation elsewhere, this shouldn't be necessary

File: public/static/js/projects.js [hunk 2]
```diff
@@ -550,6 +554,16 @@ mciModule.controller('ProjectCtrl', function ($scope, $window, $http, $location,
     delete $scope.patch_alias
   };
 
+  $scope.addWorkstationCommand = function() {
+      if (!$scope.settingsFormData.workstation_config) {
+          $scope.settingsFormData.workstation_config = [];
+      }
+      if ($scope.validWorkstationCommand($scope.workstation_command)) {
```
Comment (john-m-liu) [inline public/static/js/projects.js:561]: This should show an error if invalid. Though if there's not actually a way to get here if it's invalid, you don't need to check at all

File: service/templates/projects.html [hunk 3]
```diff
@@ -633,7 +633,49 @@ <h3>Periodic Builds</h3>
             </div>
           </div>
         </div>
+        <!-- Commands for Virtual Workstations -->
+        <div class="variables">
+          <div class="form-group">
+            <div class="col-header col-lg-6 form-control-static">
+              <h3>Virtual Workstations Commands</h3>
+              <div class="muted small">Specify a command to be run on a virtual workstation for this project, and optionally a subdirectory to run it in. If using git clone, the repo and branch will be taken from this project.</div>
```
Comment (john-m-liu) [inline service/templates/projects.html:641]: Could you say how the command will be run? It seems that it's run by the shell at startup, but I'm not sure

File: service/templates/projects.html [hunk 4]
```diff
@@ -633,7 +633,49 @@ <h3>Periodic Builds</h3>
...
+
+          <div class="form-group">
+            <div class="col-lg-4">
+              <textarea class="form-control" ng-model="workstation_command.command" placeholder="command"></textarea>
+            </div>
+            <div class="col-lg-2">
+              <input ng-model="workstation_command.directory" class="form-control" type="text" placeholder="directory">
+            </div>
+            <div class="col-lg-2">
+              <button class="plus-button btn btn-primary " ng-disabled="!workstation_command.command" type="button" ng-click="addWorkstationCommand()">
```
Comment (john-m-liu) [inline service/templates/projects.html:673]: It's pretty much the same, but should this actually use the validation function to check?

**Comment by tychoish** [comment]
Can you add a screenshot of the UI element to the review?

File: service/templates/projects.html (hunk 3 above)
Comment (tychoish) [reply service/templates/projects.html:641]: also say, commands will be: 

- run in order entered.
- all executed within the user's persistent home directory by default.

Comment (ablack12) [reply service/templates/projects.html]: Updated this description, but I can tweak it after creating the CLI command.

**Comment by ablack12** [comment]
The code from this ticket is now also part of the new EVG-7732; I'm going to close this PR but feel free to reference.

**Pull request closed without merging**